tests/test_path_analysis.py

```diff
@@ -0,0 +1,153 @@
+
+@pytest.fixture
+def data_files(ref_dir):
+    return { 
+             'pdb' : os.path.join(ref_dir, 'sim.prot.twochains.pdb'),
+             'psn' : os.path.join(ref_dir, 'sc-graph_twochains_all.dat')
+           }
+
+@pytest.fixture
+def data(data_files):
+    return pa.build_graph(data_files['psn'], data_files['pdb'])
+
+@pytest.fixture
+def source(data):
+    return pa.convert_input_to_list(user_input = "A1:A2,A57",
+                                    identifiers = data[0])
+
+@pytest.fixture
+def target(data):
+    return pa.convert_input_to_list(user_input = "B1042",
+                                    identifiers = data[0])
+
+
+@pytest.fixture
+def shortest_path(data, source, target):
+    return pa.get_shortest_paths(graph = data[1],
+                                 source = source,
+                                 target = target)
+
+@pytest.fixture
+def all_path(data, source, target):
+    return pa.get_all_simple_paths(graph = data[1],
+                                 source = source,
+                                 target = target,
+                                 maxl = 3)
+
+@pytest.fixture
+def shortest_path_graph(data, shortest_path):
+    return pa.get_persistence_graph(graph = data[1], 
+                                    paths = shortest_path, 
+                                    identifiers = data[0])
+
+@pytest.fixture
+def all_path_graph(data, all_path):
+    return pa.get_persistence_graph(graph = data[1], 
+                                    paths = all_path, 
+                                    identifiers = data[0])
+
+@pytest.fixture
+def shortest_table(data, shortest_path):
+    return pa.sort_paths(graph = data[1],
+                         paths = shortest_path,
+                         sort_by = "average_weight")
+
+@pytest.fixture
+def all_table(data, all_path):
+    return pa.sort_paths(graph = data[1],
+                         paths = all_path,
+                         sort_by = "average_weight")
+                        
+@pytest.fixture
+def metapath(data):
+    metapath = pa.get_metapath(graph = data[1],
+                               res_id = data[0],
+                               res_space = 3,
+                               node_threshold = 0.1,
+                               edge_threshold = 0.1,
+                               normalize = False)
+    metapath = pa.reorder_graph(metapath, data[0])
+    metapath = nx.to_numpy_matrix(metapath)
+    return metapath
+
+@pytest.fixture
+def metapath_norm(data):
+    metapath = pa.get_metapath(graph = data[1],
+                               res_id = data[0],
+                               res_space = 3,
+                               node_threshold = 0.1,
+                               edge_threshold = 0.1,
+                               normalize = True)
+    metapath = pa.reorder_graph(metapath, data[0])
+    metapath = nx.to_numpy_matrix(metapath)
+    return metapath
+
+
+@pytest.fixture
```

Comment:
Put all the fixtures at the beginning of the test script, followed by all test functions.